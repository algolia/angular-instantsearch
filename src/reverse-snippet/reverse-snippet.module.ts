import { NgModule } from '@angular/core';

import { NgAisReverseSnippet } from './reverse-snippet';

@NgModule({
  declarations: [NgAisReverseSnippet],
  exports: [NgAisReverseSnippet],
})
export class NgAisReverseSnippetModule {}
