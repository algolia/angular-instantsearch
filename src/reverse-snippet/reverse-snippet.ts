import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { Hit } from 'instantsearch.js';
import { reverseSnippet } from 'instantsearch.js/es/helpers';

@Component({
  selector: 'ais-reverse-snippet',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `<span class="ais-ReverseSnippet" [innerHtml]="content"></span>`,
})
export class NgAisReverseSnippet {
  @Input() private readonly attribute: string;
  @Input() private readonly hit: Partial<Hit>;
  @Input() private readonly highlightedTagName: string = 'mark';

  get content() {
    return reverseSnippet({
      attribute: this.attribute,
      hit: this.hit,
      highlightedTagName: this.highlightedTagName,
    });
  }
}
