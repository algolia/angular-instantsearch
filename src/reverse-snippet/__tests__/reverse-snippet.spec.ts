import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';

import { NgAisReverseSnippetModule } from '../reverse-snippet.module';

const render = ({ hit, attribute, highlightedTagName = 'mark' }) => {
  @Component({
    selector: 'test-component',
    template: `
      <ais-reverse-snippet [attribute]="attribute" [hit]="hit" [highlightedTagName]="highlightedTagName">
      </ais-reverse-snippet>
    `,
  })
  class TestComponent {
    hit = hit;
    attribute: string = attribute;
    highlightedTagName = highlightedTagName;
  }

  TestBed.configureTestingModule({
    declarations: [TestComponent],
    imports: [NgAisReverseSnippetModule],
  });

  const fixture = TestBed.createComponent(TestComponent);
  fixture.detectChanges();

  return fixture;
};

describe('reverse snippet', () => {
  it('should snippet strings', () => {
    const fixture = render({
      attribute: 'description',
      hit: {
        _snippetResult: {
          description: { value: '… <mark>foo</mark> bar …' },
        },
      },
    });
    expect(fixture).toMatchSnapshot();
  });

  it('should snippet with a custom highlightedTagName', () => {
    const fixture = render({
      attribute: 'description',
      hit: {
        _snippetResult: {
          description: { value: '… <mark>foo</mark> bar …' },
        },
      },
      highlightedTagName: 'em',
    });
    expect(fixture).toMatchSnapshot();
  });
});
