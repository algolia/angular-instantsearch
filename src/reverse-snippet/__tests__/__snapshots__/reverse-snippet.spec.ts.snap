// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reverse snippet should snippet strings 1`] = `
<test-component
  attribute={[Function String]}
  highlightedTagName={[Function String]}
  hit={[Function Object]}
>
  <ais-reverse-snippet>
    <span
      class="ais-ReverseSnippet"
    >
      … foo
      <mark
        class="ais-ReverseSnippet-highlighted"
      >
         bar …
      </mark>
    </span>
  </ais-reverse-snippet>
</test-component>
`;

exports[`reverse snippet should snippet with a custom highlightedTagName 1`] = `
<test-component
  attribute={[Function String]}
  highlightedTagName={[Function String]}
  hit={[Function Object]}
>
  <ais-reverse-snippet>
    <span
      class="ais-ReverseSnippet"
    >
      … foo
      <em
        class="ais-ReverseSnippet-highlighted"
      >
         bar …
      </em>
    </span>
  </ais-reverse-snippet>
</test-component>
`;
